cetmix_tower_yaml/models/cx_tower_yaml_mixin.py

```diff
@@ -110,8 +120,59 @@ def _post_process_record_values(self, values):
             values.update(
                 {"access_level": self.TO_YAML_ACCESS_LEVEL[values["access_level"]]}
             )
+        # Post process m2o fields
+        for key, value in values.items():
+            if key.endswith("_id"):
+                processed_value = self._process_m2o_value(key, value, record_mode=True)
+                values.update({key: processed_value})
+
         return values
 
+    def _process_m2o_value(self, field, value, record_mode=False):
+        """Post process many2one value
+
+        Args:
+            field (Char): Field the value belongs to
+            value (Char): Value to process
+            record_mode (Bool): If True process value as a record value
+                                else process value as a YAML value
+
+        Returns:
+            dict() or Char: record dictionary if fetch_record else reference
+        """
+
+        # Get destination model
+        if value:
+            field = self._fields.get(field)
+            if field:
+                comodel_name = field.comodel_name
+
+                # Process value
+                if comodel_name and self._model_supports_yaml(comodel_name):
+                    # Record -> Yaml
+                    if record_mode:
+                        record = self.env[comodel_name].browse(value[0])
+                        return record and record.reference
+
+                    # Yaml -> Record
+                    else:
+                        record = self.env[comodel_name].get_by_reference(value)
+                        return record and record.id
+
```

Comment:
```suggestion
        if not value:
            return False
        
        comodel_name = field.comodel_name
        
        # Process value
        if comodel_name and self._model_supports_yaml(comodel_name):
            # Record -> Yaml
            if record_mode:
                record = self.env[comodel_name].browse(value[0])
                return record and record.reference
        
            # Yaml -> Record
            else:
                record = self.env[comodel_name].get_by_reference(value)
                return record and record.id
        else:
            return False

```

Reply:
@ivs-cetmix large nested code structure

Reply:
@GabbasovDinar thank you very much for such a detailed and well documented suggestion 👍